pkg/eraser/eraser.go

```diff
@@ -118,104 +114,129 @@ func RemoveImage(ctx context.Context, client pb.ImageServiceClient, image string
 	return resp, nil
 }
 
-func contains(slice []string, str string) bool {
-	for _, s := range slice {
-		if s == str {
-			return true
-		}
-	}
-	return false
-}
-
 func removeVulnerableImages() (err error) {
-	backgroundContext, cancel := context.WithTimeout(context.Background(), Timeout)
+	backgroundContext, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
 	imageClient, conn, err := getImageClient(backgroundContext)
+	fmt.Println("received imageClient")
 
 	if err != nil {
 		return err
 	}
 
-	r, err := ListImages(backgroundContext, imageClient, "")
+	r, err := listImages(backgroundContext, imageClient, "")
 	if err != nil {
 		return err
 	}
 
 	var allImages []string
 	// map with key: sha id, value: repoTag list (contains full name of image)
-	m := make(map[string][]string)
+	idMap := make(map[string][]string)
 
 	for _, img := range r.Images {
 		allImages = append(allImages, img.Id)
-		m[img.Id] = img.RepoTags
+		idMap[img.Id] = img.RepoTags
 	}
 
 	response, err := pb.NewRuntimeServiceClient(conn).ListContainers(backgroundContext, new(pb.ListContainersRequest))
 	if err != nil {
 		return err
 	}
 
-	var runningImages []string
+	runningImages := make(map[string]struct{})
 
 	for _, container := range response.Containers {
 		curr := container.Image
-		runningImages = append(runningImages, curr.GetImage())
+		runningImages[curr.GetImage()] = struct{}{}
 	}
 
-	var nonRunningImages []string
+	nonRunningImages := make(map[string]struct{})
```

Comment:
No need to change for here, but on these map creations, since we already know the size we need, we can pre-allocate the map with `make(map[string]struct{}, len(allImages)-len(runningImages))`

Reply:
@AldairCoronel just added the `prealloc` linter, which should make it easy to point these out